crates/red_knot_test/src/db.rs

```diff
@@ -1,69 +1,60 @@
-use std::sync::Arc;
-
+use camino::{Utf8Component, Utf8PathBuf};
 use red_knot_python_semantic::lint::{LintRegistry, RuleSelection};
-use red_knot_python_semantic::{
-    default_lint_registry, Db as SemanticDb, Program, ProgramSettings, PythonPlatform,
-    SearchPathSettings,
-};
+use red_knot_python_semantic::{default_lint_registry, Db as SemanticDb};
 use ruff_db::files::{File, Files};
-use ruff_db::system::{DbWithTestSystem, System, SystemPath, SystemPathBuf, TestSystem};
+use ruff_db::system::{
+    DbWithWritableSystem, InMemorySystem, OsSystem, System, SystemPath, SystemPathBuf,
+    WritableSystem,
+};
 use ruff_db::vendored::VendoredFileSystem;
 use ruff_db::{Db as SourceDb, Upcast};
-use ruff_python_ast::PythonVersion;
+use ruff_notebook::{Notebook, NotebookError};
+use std::borrow::Cow;
+use std::sync::Arc;
+use tempfile::TempDir;
 
 #[salsa::db]
 #[derive(Clone)]
 pub(crate) struct Db {
-    project_root: SystemPathBuf,
     storage: salsa::Storage<Self>,
     files: Files,
-    system: TestSystem,
+    system: MdtestSystem,
     vendored: VendoredFileSystem,
     rule_selection: Arc<RuleSelection>,
 }
 
 impl Db {
-    pub(crate) fn setup(project_root: SystemPathBuf) -> Self {
+    pub(crate) fn setup() -> Self {
         let rule_selection = RuleSelection::from_registry(default_lint_registry());
 
-        let db = Self {
-            project_root,
+        Self {
+            system: MdtestSystem::in_memory(SystemPathBuf::from("/")),
             storage: salsa::Storage::default(),
-            system: TestSystem::default(),
             vendored: red_knot_vendored::file_system().clone(),
             files: Files::default(),
             rule_selection: Arc::new(rule_selection),
-        };
-
-        db.memory_file_system()
-            .create_directory_all(&db.project_root)
-            .unwrap();
-
-        Program::from_settings(
-            &db,
-            ProgramSettings {
-                python_version: PythonVersion::default(),
-                python_platform: PythonPlatform::default(),
-                search_paths: SearchPathSettings::new(vec![db.project_root.clone()]),
-            },
-        )
-        .expect("Invalid search path settings");
-
-        db
+        }
     }
 
-    pub(crate) fn project_root(&self) -> &SystemPath {
-        &self.project_root
+    pub(crate) fn reset_fs(&mut self) {
+        match &*self.system.0 {
```

Comment:
Nit: maybe delegate to a `self.system.reset_fs()` instead of digging so much into `MdTestSystem` internals from outside?